WordPress/src/main/java/org/wordpress/android/ui/photopicker/PhotoPickerActivity.java

```diff
@@ -328,23 +328,25 @@ public void doNext(Uri uri) {
 
     private void doMediaIdsSelected(ArrayList<Long> mediaIds, @NonNull PhotoPickerMediaSource source) {
         if (mediaIds != null && mediaIds.size() > 0) {
-            if (mBrowserType == MediaBrowserType.FEATURED_IMAGE_PICKER) {
-                // if user chose a featured image, track image picked event
-                mFeaturedImageHelper.trackFeaturedImageEvent(
-                        FeaturedImageHelper.TrackableEvent.IMAGE_PICKED,
-                        mLocalPostId
-                );
-
+            if (mBrowserType == MediaBrowserType.WP_STORIES_MEDIA_PICKER) {
+                // TODO WPSTORIES add TRACKS (see how it's tracked below? maybe do along the same lines)
                 Intent data = new Intent()
-                        .putExtra(EXTRA_MEDIA_ID, mediaIds.get(0))
+                        .putExtra(MediaBrowserActivity.RESULT_IDS, ListUtils.toLongArray(mediaIds))
+                        .putExtra(ARG_BROWSER_TYPE, mBrowserType)
                         .putExtra(EXTRA_MEDIA_SOURCE, source.name());
                 setResult(RESULT_OK, data);
                 finish();
             } else {
-                // TODO WPSTORIES add TRACKS (see how it's tracked above? maybe do along the same lines)
+                // if user chose a featured image, track image picked event
+                if (mBrowserType == MediaBrowserType.FEATURED_IMAGE_PICKER) {
```

Comment:
Oh I can see this section here is enclosed more tightly to only call the `trackFeaturedImageEvent()` method when the browserType is `FEATURED_IMAGE_PICKER` and then leave all of the rest (`else` clause) to always set a result, understood the difference now, forget the other comment 👍